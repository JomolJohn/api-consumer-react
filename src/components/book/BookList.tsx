import { Link } from "react-router-dom";
import { IBookDetail } from '../../api/bookInterface';
import './Book.css';

export interface IBookProps {
    book: IBookDetail;
}

const BookList = (props: IBookProps) => {

    return (
        <>
            <article className="card">
                <div className="card__content">
                    <h2>{props.book.title}</h2>
                    <span>{props.book.author}</span>
                    <span>{props.book.year}</span>
                    <p className="card_link"><Link to={`book/${props.book.id}`}>Read more</Link></p>
                </div>
            </article>
        </>
    );
};
export default BookList;