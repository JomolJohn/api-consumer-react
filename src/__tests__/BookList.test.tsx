import React from "react";
import { shallow } from "enzyme";
import enzymeToJson from "enzyme-to-json";
import BookList from '../components/book/BookList';

const book = {
    id: '15',
    title: 'Paul of Dune',
    year: '2008',
    author: 'Brian HerbertKevin J. Anderson',
    wiki_url: 'https://en.wikipedia.org/wiki/Paul_of_Dune',
}

describe('>>> BookList component', () => {
    it('+++ renders when one item', () => {
        jest.spyOn(React, "useState");
        const component = shallow(<BookList book={book} />);
        expect(enzymeToJson(component)).toMatchSnapshot();
    });
});