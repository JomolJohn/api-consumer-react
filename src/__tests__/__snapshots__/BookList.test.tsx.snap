// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> BookList component +++ renders when one item 1`] = `
<Fragment>
  <article
    className="card"
  >
    <div
      className="card__content"
    >
      <h2>
        Paul of Dune
      </h2>
      <span>
        Brian HerbertKevin J. Anderson
      </span>
      <span>
        2008
      </span>
      <p
        className="card_link"
      >
        <Link
          to="book/15"
        >
          Read more
        </Link>
      </p>
    </div>
  </article>
</Fragment>
`;
